packages/livekit-rtc/src/data_streams/stream_writer.ts

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2024 LiveKit, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+import type { WritableStream } from 'node:stream/web';
+
+class BaseStreamWriter<T> {
+  protected writableStream: WritableStream<T>;
+
+  protected defaultWriter: WritableStreamDefaultWriter<T>;
+
+  protected onClose?: () => void;
+
+  constructor(writableStream: WritableStream<T>, onClose?: () => void) {
+    this.writableStream = writableStream;
+    this.defaultWriter = writableStream.getWriter();
+    this.onClose = onClose;
+  }
+
+  write(chunk: T): Promise<void> {
+    return this.defaultWriter.write(chunk);
+  }
```

Comment:
possible syntactic sugar: array expansion? i feel like `([a, b])` is ugly and doesn't make things any clearer either way, so maybe being being able to do `a, b?` would be nicer. this would require some typescript witchcraft to make it not break file streams though, so i'm fine either way

Reply:
yeah, agree, I had tried that, but couldn't make it work with TypeScript :/ If you have an idea how it could work, let me know!